metadata-ingestion/src/datahub/ingestion/source/delta_lake/source.py

```diff
@@ -126,24 +137,80 @@ def create(cls, config_dict: dict, ctx: PipelineContext) -> "Source":
         config = DeltaLakeSourceConfig.parse_obj(config_dict)
         return cls(config, ctx)
 
+    inner_data_type = ""
+    counter = 0
+
+    def _get_inner_field_details(self, raw_field):
+        if "elementType" in raw_field and type(raw_field.get("elementType")) == dict:
+            self.inner_data_type = (
+                self.inner_data_type
+                + "item:"
```

Comment:
What special about `item` string here ? Is it same for struct as well as array data type ?

Reply:
will modify this to `_inner_field_details`

Reply:
in order to get the `nested` avro schema, we need to provide the `name` and `type` of the fields.
if we don't provide the `name` of the field then the UI is not generating the collapsible element of the inner field of the array.
Hence, where don't have explicit `fieldName` i.e. there I am putting `item` as a name; otherwise taking name from the column itself i.e. struct

<img width="592" alt="Screenshot 2024-02-26 at 12 55 38 pm" src="https://github.com/datahub-project/datahub/assets/158567391/2f04c573-6217-4b98-934e-2cd8ab1eaafa">